agent/consul/gateways/bind.go

```diff
@@ -65,9 +78,58 @@ func getReferences(route structs.BoundRoute) (referenceSet, gatewayRefs) {
 	gatewayRefs := make(gatewayRefs)
 	for _, ref := range route.GetParents() {
 		parentRefs[ref] = struct{}{}
-		kindName := configentry.NewKindName(structs.BoundAPIGateway, ref.Name, &ref.EnterpriseMeta)
+		kindName := configentry.NewKindName(structs.BoundAPIGateway, ref.Name, pointerTo(ref.EnterpriseMeta))
 		gatewayRefs[kindName] = append(gatewayRefs[kindName], ref)
 	}
 
 	return parentRefs, gatewayRefs
 }
+
+func requestToResourceRef(req controller.Request) structs.ResourceReference {
+	ref := structs.ResourceReference{
+		Kind: req.Kind,
+		Name: req.Name,
+	}
+	if req.Meta != nil {
+		ref.EnterpriseMeta = *req.Meta
+	}
+	return ref
+}
+
+// RemoveGateway sets the route status for the given gateway to be unbound if it should be bound
+func RemoveGateway(gateway structs.ResourceReference, entries ...structs.BoundRoute) []structs.ControlledConfigEntry {
+	now := pointerTo(time.Now().UTC())
+	modified := []structs.ControlledConfigEntry{}
+	for _, route := range entries {
+		updater := structs.NewStatusUpdater(route)
+		for _, parent := range route.GetParents() {
+			if parent.Kind == gateway.Kind && parent.Name == gateway.Name && parent.EnterpriseMeta.IsSame(&gateway.EnterpriseMeta) {
+				updater.SetCondition(structs.Condition{
+					Type:               "Bound",
+					Status:             "False",
+					Reason:             "GatewayNotFound",
+					Message:            "gateway was not found",
+					Resource:           pointerTo(parent),
```

Comment:
Is this returning a `ResourceReference` or a literal pointer? If the former, the function name could be improved

Reply:
It's a generic function, it returns a literal pointer to whatever you passed in so that you can do things like chain calls ala `pointerTo(time.Now().UTC())`, in this case it has the added benefit of acting like shadowing the `parent` loop variable so that we don't have to put something like `parent := parent` as the first line of this loop.